metadata/elasticsearch/src/main/java/com/spotify/heroic/metadata/elasticsearch/MetadataBackendKV.java

```diff
@@ -592,6 +608,14 @@ private static void buildContext(final XContentBuilder b, Series series) throws
         }
         b.endArray();
 
+        if (indexResourceIdentifiers) {
+          b.startArray(RESOURCE);
+          for (final Map.Entry<String, String> entry : series.getResource().entrySet()) {
+            b.value(entry.getKey() + TAG_DELIMITER + entry.getValue());
+          }
+          b.endArray();
+        }
+
```

Comment:
I feel that the two for loops (599 & 612) can be replaced by the same function (i.e. refactor them). AFAICT They're all the same types involved. So there'd be 2 calls to the function, which would look like :

```java
private static appendData(final XContentBuilder b, Map<String, String> data, String name) {
        b.startArray(name);
        for (var entry : data.entrySet()) {
            b.value(entry.getKey() + name + entry.getValue());
        }
        b.endArray();
}
```
then L599 would become:
`appendData(b, series.getTags(), TAGS)`

Make sense?

Reply:
+1 to this suggestion, the code is fairly similar so would be nice to make it more generic if possible